lighthouse-core/report/v2/report-generator.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const REPORT_TEMPLATE = fs.readFileSync(path.join(__dirname, './report-template.html'), 'utf8');
+// TODO: Setup a gulp pipeline to concat and minify the renderer files?
+const REPORT_JAVASCRIPT = fs.readFileSync(path.join(__dirname, './report-renderer.js'), 'utf8');
+
+class ReportGeneratorV2 {
+  /**
+   * Computes the weighted-average of the score of the list of items.
+   * @param {!Array<{score: number=, weight: number=}} items
```

Comment:
> Yeah, needing to explicitly set weight: 0 on every unscored item seemed unnecessary and always having score doesn't necessarily make sense for explicitly non-scored audits so handling the unspecified case seems reasonable.

ah yes,  that's much better

> What do you envision explicit handling would do differently?

either a comment to make it clear that the code is purposefully handling that case (`// error audit result will have a null score which will be treated as 0` or whatever) or an explicit check of `item.error` and setting the score based on the result. Either way would make it easier for future code readers to figure out the error flow or for us to change error scoring to `NaN` or whatever.